src/elements/content-sidebar/Sidebar.js

```diff
@@ -76,8 +83,32 @@ class Sidebar extends React.Component<Props, State> {
     constructor(props: Props) {
         super(props);
 
+        const hasActivity = SidebarUtils.canHaveActivitySidebar(props);
+        const hasDetails = SidebarUtils.canHaveDetailsSidebar(props);
+        const hasMetadata = SidebarUtils.shouldRenderMetadataSidebar(props, props.metadataEditors);
+        const hasSkills = SidebarUtils.shouldRenderSkillsSidebar(props, props.file);
+
+        const tablist = [];
+        if (hasActivity) {
+            tablist.push(`/${SIDEBAR_VIEW_ACTIVITY}`);
+        }
+        if (hasDetails) {
+            tablist.push(`/${SIDEBAR_VIEW_DETAILS}`);
+        }
+        if (hasSkills) {
+            tablist.push(`/${SIDEBAR_VIEW_SKILLS}`);
+        }
+        if (hasMetadata) {
+            tablist.push(`/${SIDEBAR_VIEW_METADATA}`);
+        }
+
         this.state = {
             isDirty: this.getLocationState('open') || false,
+            hasActivity,
```

Comment:
These values are currently calculated in `render` because their values are derived exclusively from props; they shouldn't live in state at all.